src/Microsoft.AspNet.Http/DefaultHttpContext.cs

```diff
@@ -124,24 +124,15 @@ public override CancellationToken RequestAborted
             set { LifetimeFeature.RequestAborted = value; }
         }
 
-        public override ISessionCollection Session
+        public override ISession Session
         {
             get
             {
-                var feature = SessionFeature;
-                if (feature == null)
-                {
-                    throw new InvalidOperationException("Session has not been configured for this application or request.");
-                }
-                if (feature.Session == null)
-                {
-                    if (feature.Factory == null)
-                    {
-                        throw new InvalidOperationException("No ISessionFactory available to create the ISession.");
-                    }
-                    feature.Session = feature.Factory.Create();
-                }
-                return new SessionCollection(feature.Session);
+                return GetSessionFeature().Session;
+            }
+            set
+            {
+                GetSessionFeature().Session = value;
```

Comment:
We don't want this to throw, we want it to create a default session feature to store the session on (if not null).